### DISCUSSION TITLE
When is the '_global' macro called?

### DISCUSSION THREAD

**Discussion by HelloThisIsFlo** [bassamanator/Sovol-SV06-firmware#64] [Q&A] [answered]
Hi,

I can't see where the `_globals` macro is called. I see it declared but never called. I also don't see it using a `delayed_gcode` instruction to automatically run that macro, so I'm wondering: When is the macro called? And consequently, when are the global variables initialized?

I'm asking because I have a custom fork with loads of modifications, I want to make sure I understand how this `_globals` macro works.

Thanks 😃

**Comment by HelloThisIsFlo** [comment, marked as answer]
Ok, as often ... I answered my own question right after posting it (despite spending 30min trying to figure out the answer prior to posting) 😅😂

So, it turns the answer is **never**. The `_globals` macro is never actually executed.

We're only _accessing the variables_ set in the macro (e.g. `printer["gcode_macro _globals"].beeping_enabled`), but never actually _running_ the `gcode` of the macro itself.

I tested this by first adding a debug message to the `gcode` block, and it was never displayed. And then I tried to remove the `gcode` block entirely ... and it still works!

So the following code is enough:

```
[gcode_macro _globals]
variable_filament_sensor_enabled: 0 # NOTE Enable(1) or disable(0) the filament sensor, if one is connected
variable_beeping_enabled: 1 # NOTE Enable(1) or disable(0) beeping everywhere except during gantry calibration
variable_bed_temp_over: 10 # NOTE Start print if bed temperature is over by this amount, otherwise wait for temperature drop
variable_pre_purge_prime_length: 1.40
gcode:
```
- No need for any `SET_GCODE_VARIABLE` instructions
- We do need the empty `gcode` block because removing the `gcode` keyword causes and error
- I tested changing variables, and the changes are reflected as expected

You may want to update this @bassamanator, it's not critical at all but considering it's a super quick fix, might be worth it. 

Either way, thanks for your amazing work it was very helpful!
Feel free to close this discussion

**Comment by bassamanator** [comment]
It seems that the code inside the `gcode:` block is in fact unnecessary. Not sure when this change occurred, or maybe I've had it wrong all along. Will consider removing it.

Appreciate you pointing this out @HelloThisIsFlo!
